internal/alloc/alloc_unix.go

```diff
@@ -47,7 +47,7 @@ func (m *mmappedMemory) Reallocate(size uint64) []byte {
 		// Commit additional memory up to new bytes.
 		err := unix.Mprotect(m.buf[com:new], unix.PROT_READ|unix.PROT_WRITE)
 		if err != nil {
-			panic(err)
+			return nil
```

Comment:
Shouldn't this one still return the error? Because now it'll move on to updating committed memory which seems incorrect?